src/middleware/authentication.middleware.ts

```diff
@@ -14,8 +14,9 @@ export const authentication = (req: Request, res: Response, next: NextFunction):
       logger.infoRequest(req, 'User not authenticated, redirecting to sign in page, status_code=302');
       return res.redirect(`/signin?return_to=${SOLD_LAND_FILTER_URL}`);
     }
-
-    logger.infoRequest(req, `User (${getLoggedInUserEmail(req.session)}) is signed in`);
+    const userEmail = getLoggedInUserEmail(req.session);
+    logger.infoRequest(req, `User (${ userEmail }) is signed in`);
+    res.locals.userEmail = userEmail;
```

Comment:
Assume this is somehow needed for sign out to work? Worth a code comment explaining why maybe as it's not immediately obvious why this is done.

Reply:
You are right, sorry, I will add a comment. I am using thehttps://expressjs.com/en/5x/api.html#res.locals to make sure that the email is available within a single request-response cycle and visible in the template. This way I will avoid changing different part of the code and is more maintainable and clean

Reply:
👍 nice. Yes, a comment would help here.

Reply:
Done